DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/ip_authoring_mmhost/README.md

```diff
@@ -0,0 +1,381 @@
+
+If no memory-mapped interface parameters is informed for a pointer, then a mmhost interface with all default parameters is assumed for it.
+```c++
+struct PointerIP{
+  int *x, *y, *z;
+  int size;
+  
+  PointerIP(int *x_, int *y_, int *z_, int size_)
+    : x(x_), y(y_), z(z_), size(size_) {}
+
+    void operator()() const {
+      for (int i = 0; i < size; ++i) {
+        int mul = x[i] * y[i];
+        int add = mul + z[i];
+        y[i] = mul;
+        z[i] = add;
+      }
+    }
+};
+```
+
+### Memory-mapped interface implementation restriction
+
+Memory-mapped interface is restricted to function implementations, which means that it is not supported for lambda implementation. Also, all parameters must be preset, even if just one parameter is being changed from the default values.
+
+#### Example Functor
+```c++
+struct MyIP {
+  mmhost(
+    ... // All mmhost parameters.
+  ) int *my_pointer;
+
+  MyIP(int *input_pointer) : my_pointer(input_pointer) { ... } // Constructor for our IP.
+    ...
+  }
+
+  void operator()() const {
+    ...
+    // Functor code
+    my_pointer[x] = y;
+    ...
+  }
+}
+```
+
+
+#### Example 1: How to set-up a correct mmhost interface
+```c++
+// A memory-mapped interface must contain all parameters, in the following order:
+  mmhost(
+    1,       // buffer_location or aspace
+    28,      // address width
+    64,      // data width
+    16,      // latency
+    1,       // read_write_mode, 0: ReadWrite, 1: Read, 2: Write
+    1,       // maxburst
+    0,       // align, 0 defaults to alignment of the type
+    1        // waitrequest, 0: false, 1: true
+  ) int *memory_mapped_pointer;
+```
+
+#### Example 2: Changing default mmhost interface.
+If the mmhost argument is specified as a conduit interface then an input wire is generated at the top-level device image module with the same width as the address width property.
+
+If the mmhost argument is specified as a register-mapped argument, then only the Avalon bus will be created at the top-level device image module. The argument will be written into the register map
+```c++
+// A memory-mapped interface that is implemented as a register.
+  register_map_mmhost(
+    1,       // buffer_location or aspace
+    28,      // address width
+    64,      // data width
+    16,      // latency
+    1,       // read_write_mode, 0: ReadWrite, 1: Read, 2: Write
+    1,       // maxburst
+    0,       // align, 0 defaults to alignment of the type
+    1        // waitrequest, 0: false, 1: true
+  ) int *memory_mapped_pointer;
+// A memory-mapped interface that is implemented as a wire.
+  conduit_mmhost(
+    1,       // buffer_location or aspace
+    28,      // address width
+    64,      // data width
+    16,      // latency
+    1,       // read_write_mode, 0: ReadWrite, 1: Read, 2: Write
+    1,       // maxburst
+    0,       // align, 0 defaults to alignment of the type
+    1        // waitrequest, 0: false, 1: true
+  ) int *memory_mapped_pointer;
+```
```

Comment:
At the end mention that when nothing is specified i.e., the macro `mmhost(...)` is used, we default to register_map based input argument interface.

Reply:
Done, at the end of the example there is a mention of that